File: TestProj/Assets/My Assets/Scripts/Modular Scripts/OnKeyDown Switch.js
//iterate through a list of disabled objects. Once we've reached the end of the list, load a scene
var myObjects : GameObject[];
var myIndex : int = 0;
var myScene : String;
function Update(){
	if(Input.anyKeyDown){
		IterateObjects();
	}
}
function IterateObjects(){
	if(myIndex < myObjects.Length-1){
		myIndex+=1;
		myObjects[myIndex-1].active = false;
		myObjects[myIndex].active = true;
	}
	else{
		Application.LoadLevel(myScene);
	}
}

File: TestProj/Assets/My Assets/Scripts/Modular Scripts/OnKeyDownSwitch.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class OnKeyDownSwitch : MonoBehaviour
{
    //iterate through a list of disabled objects. Once we've reached the end of the list, load a scene
    public GameObject[] myObjects;
    public int myIndex;
    public string myScene;
    public virtual void Update()
    {
        if (Input.anyKeyDown)
        {
            this.IterateObjects();
        }
    }

    public virtual void IterateObjects()
    {
        if (this.myIndex < (this.myObjects.Length - 1))
        {
            this.myIndex = this.myIndex + 1;
            this.myObjects[this.myIndex - 1].active = false;
            this.myObjects[this.myIndex].active = true;
        }
        else
        {
            Application.LoadLevel(this.myScene);
        }
    }

}